test/e2e/e2e_test.go

```diff
@@ -599,4 +601,31 @@ var _ = Describe("For a Kubernetes cluster with Tekton and build installed", fun
 			})
 		})
 	})
+
+	Context("when a s2i build uses a non-existent git repository as source", func() {
+		It("fails because of prompted authentication which surfaces the to the BuildRun", func() {
+			testID = generateTestID("s2i-failing")
+
+			build = createBuild(
+				testBuild,
+				testID,
+				"test/data/build_non_existing_repo.yaml",
+			)
+
+			buildRun, err = buildRunTestData(build.Namespace, testID, "test/data/buildrun_non_existing_repo.yaml")
+			Expect(err).ToNot(HaveOccurred())
+
+			validateBuildRunToFail(testBuild, buildRun)
+
+			Eventually(func() *buildv1alpha1.FailureDetails {
+				buildRun, err = testBuild.LookupBuildRun(types.NamespacedName{Name: buildRun.Name, Namespace: testBuild.Namespace})
+
+				return buildRun.Status.FailureDetails
+			}).ShouldNot(BeNil())
```

Comment:
`validateBuildRunToFail` waits until the BuildRun is failed. Imo the failureDetails should be set by then. As `buildRun` is a pointer which gets updated inside the validate function, I think that the whole `Eventually` block should go away.

Reply:
+1